subsys/tracing/CMakeLists.txt

```diff
@@ -37,6 +37,11 @@ zephyr_sources_ifdef(
   tracing_backend_ram.c
   )
 
+zephyr_sources_ifdef(
+  CONFIG_TRACING_BACKEND_RAM_RING
+  tracing_backend_ram_ring.c
```

Comment:
wouldn't it be possible with a single `tracing_backend_ram.c` file and then simply have a Kconfig setting the specifies if the ram tracing backend should execute in ring mode.

That way we don't need two implementations for the ram buffer.